### PR TITLE
.werft/VM: Fixes to improve VM stability

### PR DESCRIPTION
**Pull request by ArthurSens**
#### 1
For some unknown reason, the kubectl context is set to `werft` inside out jobs. @meysholdt and I tried to debug but could find any trace of it written in kubeconfig files.

To make sure we don't use this namespace, were explicitly setting the namespace when checking for the rollout of server component.

#### 2

Also, sweeper is not implemented in a way that works for Harvester VMs, so we're skipping its deployment since it runs in the namespace werft, which doesn't exist and causes the job to fail.

#### 3

Logs are still telling us that `kubectl --kubeconfig=${KUBECONFIG_PATH} -n ${namespace} port-forward service/proxy 6443:6443` fails at the end of the job.

`killall` abruptly kills all kubectl commands running asynchronously inside our job, I've added the `dontCheckRc` option so the job doesn't fail after they get killed.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by mads-hartmann** [comment]
@ArthurSens Right now it is expected that the kubectl port-forward fails at the end of the job - we terminate it [here](https://github.com/gitpod-io/gitpod/blob/main/.werft/build.ts#L40). As long as it doesn't cause the Werft job to fail (and it shouldn't) it isn't a problem :) We could make it a bit nicer and specify a signal in the `killall` command which causes it to gracefully shutdown instead - but I don't think it's worth the effort as we're really hoping to remove the port-forwarding when implement the following issues

- [Add SSH proxy to Harvester for easy SSH access to VM](https://github.com/gitpod-io/harvester/issues/16)
- [Use kube-api proxy instead of port-forwarding](https://github.com/gitpod-io/gitpod/issues/7519)

**Comment by ArthurSens** [comment]
> @ArthurSens Right now it is expected that the kubectl port-forward fails at the end of the job - we terminate it [here](https://github.com/gitpod-io/gitpod/blob/main/.werft/build.ts#L40). As long as it doesn't cause the Werft job to fail (and it shouldn't) it isn't a problem :) 

The problem is that it **_is_** apparently making the job fail 😕

**Comment by mads-hartmann** [comment]
@ArthurSens Oh 😝 if you give me a lnk to a failing job I'd be happy to take a look.

**Comment by ArthurSens** [comment]
Sure thing, here it is :)

https://werft.gitpod-dev.com/job/gitpod-build-as-vm-stability.4/raw

**Comment by ArthurSens** [comment]
Alright, got it working finally.

Earlier we were discussing "why it fails if we have `|| true` at the end of the killall command?" Well, that command in particular indeed didn't fail. However, the processes killed by it were the ones failing the job.

I've added the `dontCheckRc` option and now the job doesn't fail :)

**Review by mads-hartmann — APPROVED** [review]
Nice catch, LGTM

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: b2f05b38f445f9903e5858c523142f2a1ebfb6e3</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7505#pullrequestreview-848850478" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7505#pullrequestreview-848850478" title="Approved">mads-hartmann</a>*

*No associated issue*. Requirement bypassed by manually added approval.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
